src/main/java/duke/Command.java

```diff
@@ -0,0 +1,26 @@
+package duke;
+
+/**
+ * Represents an abstract Command class.
+ *
+ * @author Sherman Ng Wei Sheng
+ */
+public abstract class Command {
+
+    /**
+     * Returns true if the command is a programme terminating command.
+     *
+     * @return True if it is a terminating command and false otherwise.
+     */
+    public abstract boolean isExit();
```

Comment:
Perhaps the default value returned by isExit() can be false? Assuming there will be fewer terminating commands than non-terminating ones. Then in the terminating commands, this function will be overridden, otherwise there is no need to override.